web/src/api/api.ts

```diff
@@ -114,35 +125,31 @@ async function fetchMetadata(
   return await d3.csv(url);
 }
 
-async function fetchFragments(
-  pairId: number,
-  url = DATA_URL + "fragments/"
-): Promise<string> {
-  return await d3.text(url + pairId + ".json");
-}
-
 function parseFiles(fileData: d3.DSVRowArray): ObjMap<File> {
   return Object.fromEntries(
     fileData.map(row => {
       const extra = JSON.parse(row.extra || "{}");
       extra.timestamp = extra.createdAt && new Date(extra.createdAt);
       row.extra = extra;
+      row.mapping = JSON.parse(row.mapping || "[]");
+      row.ast = JSON.parse(row.ast || "[]");
```

Comment:
I'm not 100% sure, but this might be cause the UI to hang for a while when loading a lot of files (parsing JSON is notoriously slow). Can you check if this is the case?

If this is indeed causing the UI to hang, we might consider delaying the `JSON.parse` to when a file is actually needed.